syntax = "proto3";

package otservice;

message VideoRequest {
  bytes video_file = 1;
  string model = 2;
}

message VideoResponse {
  bool success = 1;  // Whether the upload was successful
  string message = 2;  // A response message from the server
}


message LogEntry {
  string ip_address = 1;
  double grpc_system_response_time = 2;
  double total_response_time = 3;
}


message LogResponse {
  bool success = 1;
  string message = 2;
}

service OtService {
  // Upload the video file using client streaming.
  rpc UploadVideo(stream VideoRequest) returns (VideoResponse) {}
  
  rpc SendLogEntry(LogEntry) returns (LogResponse) {}
}